packages/graph-engine/src/utils/node.ts

```diff
@@ -12,3 +12,22 @@ export const getAllOutputs = <T extends Node>(node: T) => {
 		Object.entries(node.outputs).map(([key, value]) => [key, value.value])
 	) as UnwrapOutput<T['outputs']>;
 };
+
+export function cloneInnerGraph(originalNode: Node, clonedNode: Node) {
+	const original = originalNode as any;
+	const cloned = clonedNode as any;
+
+	if (
+		original._innerGraph &&
+		typeof original._innerGraph.clone === 'function'
```

Comment:
This is still always assuming the accessor path and that there will only be a single graph